### PR TITLE
fix: stop adding unnecessary null terminators to strings

### PR DESCRIPTION
**Pull request by voxeljorge**
When using this project with AWS Kinesis Video, Kinesis Video fails to parse the CodecID field of an MKV TrackEntry due to the fact that this package guarantees that strings will always have null terminators.

According to the EBML spec, null termination is allowed but not recommended except in the case where a field is being overwritten in place, such that the subsequent data in a string value can be ignored without fully overwriting it (in case the field is very large for example).

https://www.rfc-editor.org/rfc/rfc8794.html#terminating-elements

While the behavior in this package is technically correct, for better compatibility it is advisable not to null terminate strings unless absolutely necessary.

### CODE REVIEW

**Comment by at-wat** [comment]
Looks like some tests aren't updated

**Comment by voxeljorge** [comment]
I'm happy to go through and update more tests, do you have any feedback on the change itself though?

**Comment by at-wat** [comment]
I think the change itself is reasonable

**Comment by voxeljorge** [comment]
@at-wat Sorry for the long delay here but I think I've fixed the broken test, at least `go test ./...` passes now.

File: marshal_test.go [hunk 1]
```diff
@@ -179,9 +179,9 @@ func TestMarshal(t *testing.T) {
...
 				{
-					0x15, 0x49, 0xA9, 0x66, 0x90,
-					0x4D, 0x80, 0x85, 0x74, 0x65, 0x73, 0x74, 0x00,
-					0x57, 0x41, 0x85, 0x61, 0x62, 0x63, 0x64, 0x00,
+					0x15, 0x49, 0xA9, 0x66, 0x8E,
+					0x4D, 0x80, 0x84, 0x74, 0x65, 0x73, 0x74,
+					0x57, 0x41, 0x84, 0x61, 0x62, 0x63, 0x64,
 				},
 				{ // Go map element order is unstable
 					0x15, 0x49, 0xA9, 0x66, 0x90,
```
Comment (at-wat) [inline marshal_test.go:187]: Data below should be updated as well

Comment (voxeljorge) [reply marshal_test.go]: fixed

File: value.go [hunk 2]
```diff
@@ -358,8 +358,8 @@ func encodeString(i interface{}, n uint64) ([]byte, error) {
 	if !ok {
 		return []byte{}, wrapErrorf(ErrInvalidType, "writing %T as string", i)
 	}
-	if uint64(len(v)+1) >= n {
-		return append([]byte(v), 0x00), nil
+	if uint64(len(v)) >= n {
+		return append([]byte(v)), nil
```
Comment (at-wat) [inline value.go:362]:
```suggestion
		return []byte(v), nil
```

**Review by at-wat — APPROVED** [review]
LGTM

**Pull request merged**